upcloud/account.go

```diff
@@ -247,3 +248,37 @@ type BillingResourceDetail struct {
 	BillableSizeGiB int     `json:"billable_size_gib,omitempty"`
 	Labels          []Label `json:"labels,omitempty"`
 }
+
+func (brd *BillingResourceDetail) UnmarshalJSON(data []byte) error {
```

Comment:
I modified this to use custom `UnmarshalJSON`: I feel like your original solution with the custom type was cleaner, but I would prefer to keep the types in the exported struct unchanged for now 🤔